From e1672162696e38a5da1f8a4e4c5d7f58a1c3fe9f Mon Sep 17 00:00:00 2001
From: Tong Tiangen <tongtiangen@huawei.com>
Date: Fri, 13 Jan 2023 07:23:43 +0000
Subject: [PATCH] arm64: fix return value type of memcpy_mcs()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e1672162696e38a5da1f8a4e4c5d7f58a1c3fe9f
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I6AQE8
CVE: NA

-------------------------------

The return value type of function memcpy_mcs() should be unsigned long.
Currently, using this function will force type conversion, but the error
of the return value type must be corrected.

Fixes: ee7b5d7559e6 ("arm64: introduce copy_mc_to_kernel() implementation")
Signed-off-by: Tong Tiangen <tongtiangen@huawei.com>
(cherry picked from commit 4a724b4c47e52fe67be5cb5818adb68aa9f0361a)
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/include/asm/string.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/include/asm/string.h b/arch/arm64/include/asm/string.h
index 08e0327a1719..c12a3ca42d61 100644
--- a/arch/arm64/include/asm/string.h
+++ b/arch/arm64/include/asm/string.h
@@ -36,8 +36,8 @@ extern void *memcpy(void *, const void *, __kernel_size_t);
 extern void *__memcpy(void *, const void *, __kernel_size_t);
 
 #define __HAVE_ARCH_MEMCPY_MC
-extern unsigned long *memcpy_mcs(void *, const void *, __kernel_size_t);
-extern unsigned long *__memcpy_mcs(void *, const void *, __kernel_size_t);
+extern unsigned long memcpy_mcs(void *, const void *, __kernel_size_t);
+extern unsigned long __memcpy_mcs(void *, const void *, __kernel_size_t);
 
 #define __HAVE_ARCH_MEMMOVE
 extern void *memmove(void *, const void *, __kernel_size_t);
-- 
2.33.0

